x-pack/legacy/plugins/siem/public/pages/detection_engine/signals/index.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFilterButton, EuiFilterGroup } from '@elastic/eui';
+import React, { useCallback, useState } from 'react';
+import { OpenSignals } from './components/open_signals';
+import { ClosedSignals } from './components/closed_signals';
+import { GlobalTime } from '../../../containers/global_time';
+import { StatefulEventsViewer } from '../../../components/events_viewer';
+import * as i18n from './translations';
+import { DEFAULT_SIGNALS_INDEX } from '../../../../common/constants';
+import { signalsDefaultModel } from './default_model';
+
+const SIGNALS_PAGE_TIMELINE_ID = 'signals-page';
+const FILTER_OPEN = 'open';
+const FILTER_CLOSED = 'closed';
+
+export const SignalsTableFilterGroup = React.memo(
+  ({ onFilterGroupChanged }: { onFilterGroupChanged: (filterGroup: string) => void }) => {
+    const [filterGroup, setFilterGroup] = useState(FILTER_OPEN);
+
+    return (
+      <EuiFilterGroup>
+        <EuiFilterButton
+          hasActiveFilters={filterGroup === FILTER_OPEN}
+          onClick={() => {
+            setFilterGroup(FILTER_OPEN);
+            onFilterGroupChanged(FILTER_OPEN);
+          }}
+          withNext
+        >
+          {'Open signals'}
+        </EuiFilterButton>
+
+        <EuiFilterButton
+          hasActiveFilters={filterGroup === FILTER_CLOSED}
+          onClick={() => setFilterGroup(FILTER_CLOSED)}
+        >
+          {'Closed signals'}
+        </EuiFilterButton>
+      </EuiFilterGroup>
+    );
+  }
+);
+
+export const SignalsTable = React.memo(() => {
+  const [filterGroup, setFilterGroup] = useState(FILTER_OPEN);
+
+  const onFilterGroupChangedCallback = useCallback(
+    (newFilterGroup: string) => {
+      setFilterGroup(filterGroup);
+    },
+    [setFilterGroup]
```

Comment:
should `filterGroup` be a dependency here?

Reply:
No, but it should be 
```` ts 
setFilterGroup(newFilterGroup)
```